src/NetSuiteClient.php

```diff
@@ -75,20 +81,35 @@ public function setDataCenterUrl(array $config)
 
     public static function createFromEnv($options = array(), $client = null)
     {
+        if (!isset($_SERVER['APP_ENV'])) {
```

Comment:
I'm feeling like this whole block from 84-93 should be omitted. My reasoning is that if the user wishes to rely on env-based config loading, then they are free to accomplish this in any way they desire. There are at least a few libraries out there and they could home brew something or even hard code something in their apache configuration to populate the $ENV superglobal. I feel that introducing knowledge of specific external non-dependent libraries in this context is unnecessary and that a preferable solution is to just use ENV data if it exists, and don't use it if it doesn't. Then expand the readme with a brief note explaining the suggested packages to accomplish env loading and allow the user to decide how to accomplish this. It means no having to maintain this block in the future if/when the Dotenv gets moved into a new namespace, renamed, or replaced by some more ubiquitous library.

I'm happy to be convinced that there's more value to this than I'm considering. If so, change my mind!

Reply:
It was more about trying to ensure that we handle there being no config at all.
The two libraries cover Symfony and Laravel primarily as well as being the two most widely used env loading libraries for PHP, and covering them means that for a lot of developers you're going to be up and running with less faffing about, but it's not stopping the developer from populating getenv in a different way, or simply passing a $config through to the library as used to happen.

I'm happy to remove it if you feel it's complexity > value but IMO handling the most common methods of populating APP_ENV is useful.

Reply:
Hi again, sorry for the delay. A busy weekend to cap off an eventful week.

I agree with you that this library should include intelligence to detect the absence of a usable configuration and then communicate this to the user at the library level (instead of just letting the ensuing soap calls fail). I also agree that a RuntimeException is suitable.

That said, I still have reluctance about infusing more logic than necessary into this library to do this. I already talked about the explicit references to external, non-required libraries, but looking at this again I actually have a greater concern with introducing code to actually do the loading of env configuration from a specific file. For instance, it's possible for a user to choose not to install composer libraries into `./vendor`. Someone might prefer to use `./contrib/libraries` or any other arbitrary path. This would break the functionality which assumes that we can load a `.env` file from one directory above where our library is stored.

So I definitely agree we ought to look for configuration data and then throw an exception when we can't find any, neither passed in nor in an already-loaded set of ENV data, but I still believe that for a few reasons we should remove the explicit library checks and especially the auto-loading of a file. Since this is a library and not a standalone package I don't think there needs to be any expectation by the user that this should load env data on its own in addition to what their framework might already be doing in usual cases. I don't mean to belabor this unnecessarily but I am just not feeling that the relatively small value added by this extra functionality (given we both agree that almost everyone is using this inside a framework or some platform already leveraging dotenv type functionality), is justified by the need to then maintain it and, probably more significantly, the need to support users who get confused by the various ways configs can be implicitly and explicitly loaded into the NetSuiteClient.

If we can reach an agreement here, I think all that needs to be changed is to simplify `createFromEnv()`, removing the entire first block of code doing the checks. Then in the constructor, throw the runtime exception when we have tried to look at the `$config` array and then tried to build one with `createFromEnv()` and still don't have valid config data.